tests/main/snap-interface/task.yaml

```diff
@@ -2,10 +2,13 @@ summary: Check that "snap interface" works as expected
 details: |
     The "snap interface" command displays a listing of used interfaces
 execute: |
-    snap interface | MATCH 'core-support\s+ special permissions for the core snap'
```

Comment:
It looks like this PR is removing the testing of the core-support interface, without removing the fact that this interface exists and has logic inside it today. We should either remove the interface, or not remove its tests I think?  It's very suspect otherwise.. either we don't need the interface to exist or we do.

Reply:
This doesn't remove the interface, it removes assumption that a plug exists on core.
We cannot remove the interface entirely (see https://github.com/snapcore/snapd/pull/5411 - we found that this interface is abused by branded stores) but we want to remove our use of it from the core snap (which is indeed safe as it is not really used by snapd in practice).

Reply:
Right, exactly what the point above is about: this is not removing the interface. It is removing the testing of the interface. Either we need the interface in place, or we don't. We shouldn't remove the testing of the interface if we need it in place.

Reply:
This is in no way removing of the testing of the interface. This removes the assumption that it always exists on the core snap. The output of `snap interface` is limited to those interfaces that have plugs or slots in the system. Since we removed `core-support-plug` from the `core` snap and since it never existed on `core18` we must adjust this test accordingly.